### DISCUSSION TITLE
scCO2 dissolving in brine (Error in phreeqc part)

### DISCUSSION THREAD

**Discussion by Mohamadreza1998** [GeoChemFoam/GeoChemFoam#9] [Q&A] [answered]
Hi, thanks for your perfect package to simulate reactive transport.
I assumed a porous media geometry for simulation of  dissolution of scCO2 to brine that injected scCO2 from left side with 954.305 mol/m3 and assumed the brine in the internal geometry with pH=7 (H+, OH-=1e-7 mol/m3). After, I set all parameter and run the simulation with interReactiveTransferFoam solver then the simulation has error. The error from phreeqc. The error is:

========================================================================
WARNING: Trying reduced tolerance 1e-18 ...

WARNING: Maximum iterations exceeded, 200

WARNING: Numerical method failed with this set of convergence parameters.

WARNING: The program has failed to converge to a numerical solution.

The following equations were not satisfied:
ERROR:               A(H2O) Activity of water has not converged. 	Residual: 2.167989e-02

ERROR: Numerical method failed on all combinations of convergence parameters, cell/soln/mix 0
Stopping.
ERROR: ERROR:               A(H2O) Activity of water has not converged. 	Residual: 2.167989e-02

ERROR: Numerical method failed on all combinations of convergence parameters, cell/soln/mix 0

ERROR:               A(H2O) Activity of water has not converged. 	Residual: 2.167989e-02

ERROR: Numerical method failed on all combinations of convergence parameters, cell/soln/mix 0

ERROR: PhreeqcRM failed.
ERROR: PhreeqcRM::RunCells
========================================================================

The thermoPhysicalProperties is default of tutorial.
I was changed the concentration of CO2 to 10 mole/m3 and the simulation is not any error but this concentration is very small and not suitable then tried with 100 mol/m3, 500 mol/m3 and the simulation has error again.

I'm appreciate to help me to solving this error.
Thanks

**Comment by GeoChemFoam** [comment]
Hi,

This is a very difficult problem. It would be great to have some more information about your set up. Is the concentration equilibrated initially? Are any other parameters different from the tutorial?

This error is coming from PHREEQC which may indicate that the chemical equilibrium is at a very very low pH. You may need to adjust them to more realistic values, refine the grid, increase the residual tolerance, or decrease the timestep.

It is hard to know without more information.

Cheers,

Hannah

On 22 May 2023, at 14:39, Mohamadreza1998 ***@***.***> wrote:

****************************************************************
Caution: This email originated from a sender outside Heriot-Watt University.
Do not follow links or open attachments if you doubt the authenticity of the sender or the content.
****************************************************************



Hi, thanks for your perfect package to simulate reactive transport.
I assumed a porous media geometry for simulation of dissolution of scCO2 to brine that injected scCO2 with 954.305 mol/m3 and assumed the brine internal geometry with pH=7 (H+, OH-=1e-7 mol/m3). After, I set all parameter and run the simulation with interReactiveTransferFoam solver then the simulation has error. The error from phreeqc. The error is:

========================================================================
WARNING: Trying reduced tolerance 1e-18 ...

WARNING: Maximum iterations exceeded, 200

WARNING: Numerical method failed with this set of convergence parameters.

WARNING: The program has failed to converge to a numerical solution.

The following equations were not satisfied:
ERROR: A(H2O) Activity of water has not converged. Residual: 2.167989e-02

ERROR: Numerical method failed on all combinations of convergence parameters, cell/soln/mix 0
Stopping.
ERROR: ERROR: A(H2O) Activity of water has not converged. Residual: 2.167989e-02

ERROR: Numerical method failed on all combinations of convergence parameters, cell/soln/mix 0

ERROR: A(H2O) Activity of water has not converged. Residual: 2.167989e-02

ERROR: Numerical method failed on all combinations of convergence parameters, cell/soln/mix 0

ERROR: PhreeqcRM failed.
ERROR: PhreeqcRM::RunCells

I was changed the concentration of CO2 to 10 mole/m3 and the simulation is not any error but this concentration is very small and not suitable then tried with 100 mol/m3, 500 mol/m3 and the simulation has error again.

I'm appreciate to help me to solving this error.
Thanks

—
Reply to this email directly, view it on GitHub<https://github.com/GeoChemFoam/GeoChemFoam/discussions/9>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AWDOAFUH64OHJNLTCNRJHUTXHNUBPANCNFSM6AAAAAAYKO6XBA>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

________________________________

Founded in 1821, Heriot-Watt is a leader in ideas and solutions. With campuses and students across the entire globe we span the world, delivering innovation and educational excellence in business, engineering, design and the physical, social and life sciences. This email is generated from the Heriot-Watt University Group, which includes:

  1.  Heriot-Watt University, a Scottish charity registered under number SC000278
  2.  Heriot- Watt Services Limited (Oriam), Scotland's national performance centre for sport. Heriot-Watt Services Limited is a private limited company registered is Scotland with registered number SC271030 and registered office at Research & Enterprise Services Heriot-Watt University, Riccarton, Edinburgh, EH14 4AS.

The contents (including any attachments) are confidential. If you are not the intended recipient of this e-mail, any disclosure, copying, distribution or use of its contents is strictly prohibited, and you should please notify the sender immediately and then delete it (including any attachments) from your system.

**Reply by Mohamadreza1998** [reply]
Hi, Dr. Hannah Menke.

I send the simulation files to your email.

With best wishes

Mohamadreza

**Comment by Santhosh91000** [comment]
Hello Hannah, Mohamad,

I am encountering the same type of issues, which occurs when I try to refine my mesh. Meaning that with a coarse mesh, everything works fine but as I refine it too much, this error pops out.  
Did you guys find a solution for what is potentially happening here ?

Thanks for your time!
Sincerely,
Santhosh

**Reply by hannahmenke** [reply]
Hi Santhosh, 

Have you tried the suggestions in the previous post? ie. lowering the timestep or changing residual tolerance?

We do not provide support for phreeqc errors so I cannot be much help here.

Best wishes,

Hannah

**Reply by Santhosh91000** [reply]
Hello Hannah,

Thanks for the reply! I actually tried it with values of diffusion coefficient and concentration values not too precise (meaning rounded to a tenth or unity) and it seems to do the trick. I think it is related to issue with flotation coding maybe ?

I have a follow-up question. I am doing simulation of CO2 carried by water in cavity (in diverse form, gaseous, liquid or supercritical) but in 3D. The simulation is really long to converge, especially because I am using a low courant number (around 0.25 all the way down to 0.005) and I am also trying to use a fine uniform meshing (1-2 micro-meter per cell). Do you have any hints or advice on how I can speed up the computational time, while keeping the results correct (avoid spurrious currents) ?

Thanks again for your time!
Sincerely,
Santhosh

**Comment by julienmaes** [comment]
Are you using a maxDeltaT that satisfies the Brackbill condition?

**Reply by Santhosh91000** [reply]
Hello julien,

Yes! That is the main issue. So I am using ''adjustableRuntime'' to make sure that the time step respects the Courant number. I found that I need a Courant number of 0.015 for a uniform meshing of 2 microm/cell selected (because it is about the size of the thin liquid film) to respect the Brackbill condition. Moreover, to respect the Galusinski and Vigneaux criterion that take into account the viscous effect, I would need a Courant number of 0.005.
However, the simulation is in 3D and it takes a really long time to even have one time step. Actually I cannot even have one time step after 4-5 days of running simulation, I need to increase the Courant number, to have results.
This is the point at which I find myself encountering a challenge.

**Comment by GeoChemFoam** [comment]
Hi Santhosh,

Unfortunately as far as we are aware, the Brackbill constraint cannot be violated.

However, if you find a way around it, please let us know!

Best wishes,

Hannah

**Comment by Santhosh91000** [comment, marked as answer]
Hello Hannah, Julien,

thanks for the time and help! I'll see what I can do on my side. 


Sincerely,
Santhosh
